bee-pow/benches/pow.rs

```diff
@@ -36,21 +36,27 @@ fn mining_benchmark(c: &mut Criterion) {
     g.measurement_time(Duration::from_secs(120));
     g.throughput(Throughput::Elements(1));
 
-    g.bench_function("random message (single-threaded)", |b| {
+    g.bench_function("rando message (single-threaded)", |b| {
         let miner = MinerBuilder::default().with_num_workers(1).finish();
         b.iter_batched(
-            // TODO: Something is odd here `rand_message` takes forever.
-            rand::random::<[u8; 32]>, 
-            |bytes| miner.nonce(&bytes, MINIMUM_POW_SCORE),
+            || rand_message().pack_new(),
+            |bytes| miner.nonce(&bytes[..bytes.len() - std::mem::size_of::<u64>()], MINIMUM_POW_SCORE),
             BatchSize::SmallInput,
         )
     });
 
     g.bench_function("random message (multi-threaded)", |b| {
         let miner = MinerBuilder::default().with_num_workers(num_cpus::get()).finish();
         b.iter_batched(
-            rand::random::<[u8; 32]>,
-            |bytes| miner.nonce(&bytes, MINIMUM_POW_SCORE),
+            // TODO why is `rand::random::<[u8; 32]>` so much faster?
```

Comment:
I think it has to do with the fact that `pack_to_vec` has to do a memory allocation every single time.

Reply:
`pack_new()` get's called by the setup closure, so I don't think it factors into the timings. It probably boils down to the size of the message in bytes, as that influences the required PoW (thanks @thibault-martinez).